src/components/App/App.js

```diff
@@ -41,24 +33,28 @@ class App extends Component {
     return (
       <main className="app">
         <Nav />
-        <SearchBar searchCities={ this.searchCities } />
-          {this.state.error &&
-            <h2>{ this.state.error }</h2>
-          }
-          {this.state.isLoading &&
+          {this.state.error && <h2>{ this.state.error }</h2>}
+          {!this.state.error && !this.state.cities.length &&
             <img className="loading-icon" src={loadingIcon} alt="page is loading" />
           }
           {this.state.cities &&
             <Switch>
               <Route exact path='/'>
-              {this.state.searchedCities.length ?
-                <Cities cities={ this.state.searchedCities } /> :
-                <Cities cities={ this.state.cities } />
-              }
+              {this.state.searchedCities.length 
+                ?
+               <>
+                <SearchBar searchCities={ this.searchCities } />
+                <Cities cities={ this.state.searchedCities } /> 
+               </>
+                :
+                <>
+                <SearchBar searchCities={ this.searchCities } />
+                <Cities cities={ this.state.cities } /> 
+                </> }
```

Comment:
Glad that we've got this logic sorted.  In the future, it would probably make sense to have separate components 'AllCities' & 'FilteredCities', and render the search bar within each component.  React architecture is tricky!